x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -115,7 +123,29 @@ static DeprecationIssue checkUniqueRealmOrders(final Settings settings, final Pl
         );
     }
 
-    static DeprecationIssue checkImplicitlyDisabledBasicRealms(final Settings settings, final PluginsAndModules pluginsAndModules) {
+    static DeprecationIssue checkImplicitlyDisabledSecurityOnBasicAndTrial(final Settings settings,
+                                                                           final PluginsAndModules pluginsAndModules,
+                                                                           final XPackLicenseState licenseState) {
+        if ( XPackSettings.SECURITY_ENABLED.exists(settings) == false
+            && (licenseState.getOperationMode().equals(License.OperationMode.BASIC)
+            || licenseState.getOperationMode().equals(License.OperationMode.TRIAL))) {
+          String details = "The default behavior of disabling security on " + licenseState.getOperationMode().description()
+              + " licenses is deprecated. A later version of Elasticsearch will set [xpack.security.enabled] to \"true\" "
```

Comment:
Could this be better phrased as "will change the default of of [xpack.security.enabled] to 'true' regardless of the license level."?

Reply:
Thanks for the suggestion Ryan. I don't have a strong opinion here, I think both versions convey the meaning equally clearly and as a non-native speaker, I can't say I see an impactful difference. Your version is shorter and I do like that. I'll ping @lockewritesdocs for an opinion and if he's also +1, I'll make the change

Reply:
I think that "will change the default of [xpack.security.enabled] to 'true'" is a little ambiguous. It almost sounds like we're moving away from the default, which is currently  `[xpack.security.enabled]` to `true`. I'd suggest the following revision, which incorporates Ryan's shorter version:

>A later version of Elasticsearch will set `[xpack.security.enabled]` to `true` by default, regardless of the license level.

Reply:
I was trying to move away from "will set" because it implies the value is actually being set (for example in elasticsearch.yml) when it is not. It is a default value that we will now be applied regardless of other factors.

Reply:
That's a good callout @rjernst. The behavior will operate as if `[xpack.security.enabled]` is `true`. Perhaps we can explain that nuance with something like:

>A later version of Elasticsearch will enable security by default and behave as if `[xpack.security.enabled]` is `true`, regardless of the license level.

Reply:
I don't think we're using phrases such as "The behavior will operate as if X is Y" for settings that have default value set to Y. I find this confusing to be honest. May I suggest 

`A later version of Elasticsearch will set [xpack.security.enabled] to \"true\"`

-->

`In a later version of Elasticsearch, the value of [xpack.security.enabled] will default to \"true\"` ?